### PR TITLE
Track Admin Game Time

### LINKED ISSUE #24404: Track Time While Adminned
## Description

SSL was mentioned on [discord](https://discord.com/channels/310555209753690112/1182777817105190982/1190310058714738841), should only track playtime while adminned and should stop when deadminned. This should not be show in the regular playtime tracker unless that user is an admin.

### PR DESCRIPTION
**Pull request by Titian3**
## About the PR
Added time tracking for when a player is an admin mode.

## Why / Balance
This was requested in the admin feature request and also as a potential way to help onboard and review new admins on to the team. Fixes #24404 

## Technical details
Add a new time tracker for when an player is an admin and disables when de-admin'ing

## Media
image

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
NA

**Changelog**
:cl:
ADMIN:
- add: Added admin playtime tracking and included it in overall playtime tracking.

### CODE REVIEW

File: Content.Server/Players/PlayTimeTracking/PlayTimeTrackingSystem.cs [hunk 1]
```diff
@@ -47,6 +50,7 @@ public override void Initialize()
         SubscribeLocalEvent<UnAFKEvent>(OnUnAFK);
         SubscribeLocalEvent<MobStateChangedEvent>(OnMobStateChanged);
         SubscribeLocalEvent<PlayerJoinedLobbyEvent>(OnPlayerJoinedLobby);
+        _adminManager.OnPermsChanged += AdminPermsChanged;
     }
```
Comment (metalgearsloth) [inline Content.Server/Players/PlayTimeTracking/PlayTimeTrackingSystem.cs:54]: Unsubscribe on shutdown.

File: Content.Shared/Players/PlayTimeTracking/PlayTimeTrackingShared.cs [hunk 2]
```diff
@@ -6,4 +6,9 @@ public static class PlayTimeTrackingShared
     /// The prototype ID of the play time tracker that represents overall playtime, i.e. not tied to any one role.
     /// </summary>
     public const string TrackerOverall = "Overall";
+
+    /// <summary>
+    /// The prototype ID of the play time tracker that represents admin time, when a player is in game as admin.
+    /// </summary>
+    public const string TrackerAdmin = "Admin";
```
Comment (metalgearsloth) [inline Content.Shared/Players/PlayTimeTracking/PlayTimeTrackingShared.cs:13]: Use validateprototype.

File: Content.Server/Players/PlayTimeTracking/PlayTimeTrackingSystem.cs [hunk 3]
```diff
@@ -131,6 +140,12 @@ private void OnAFK(ref AFKEvent ev)
         _tracking.QueueRefreshTrackers(ev.Session);
     }
 
+    private void AdminPermsChanged(AdminPermsChangedEventArgs admin)
+    {
+        _tracking.QueueRefreshTrackers(admin.Player);
+    }
+
+
```
Comment (metalgearsloth) [inline Content.Server/Players/PlayTimeTracking/PlayTimeTrackingSystem.cs:148]:
```suggestion
    }

```

File: Content.Server/Players/PlayTimeTracking/PlayTimeTrackingSystem.cs [hunk 4]
```diff
@@ -58,6 +62,11 @@ public override void Shutdown()
 
     private void CalcTrackers(ICommonSession player, HashSet<string> trackers)
     {
+        if (_adminManager.IsAdmin(player))
+        {
+            trackers.Add(PlayTimeTrackingShared.TrackerAdmin);
+        }
+
         if (_afk.IsAfk(player))
```
Comment (metalgearsloth) [inline Content.Server/Players/PlayTimeTracking/PlayTimeTrackingSystem.cs:70]: Is it a good idea to bump admin playtime above overall. If you really want to bypass it for afk seems like you should do it at an admin level.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Content.Server/Players/PlayTimeTracking/PlayTimeTrackingSystem.cs (hunk 4 above)
Comment (Titian3) [reply Content.Server/Players/PlayTimeTracking/PlayTimeTrackingSystem.cs:71]: I'm not really sure what you mean by this one sorry, I'll ping you tomorrow morning when I'm not half asleep for more info. 😴 
Thank you for the reviews though appreciate it!

Comment (metalgearsloth) [reply Content.Server/Players/PlayTimeTracking/PlayTimeTrackingSystem.cs:71]: Admin playtime gets bumped before afk does which means admin time can be higher than their playtime overall which I'm not a big fan of.

File: Content.Shared/Players/PlayTimeTracking/PlayTimeTrackingShared.cs (hunk 2 above)
Comment (Titian3) [reply Content.Shared/Players/PlayTimeTracking/PlayTimeTrackingShared.cs]: `[ValidatePrototypeId<PlayTimeTrackerPrototype>]
    public const string TrackerAdmin = "Admin";`
    
 Added this one in.

**Comment by Titian3** [comment]
@metalgearsloth Should be all the requested changes updated now. 🚀

**Review by ShadowCommander — APPROVED** [review]

**Comment by ShadowCommander** [comment]
Going to move the changelog to the admin changelog

**Pull request merged**